modules/core/src/lib/attribute/attribute.ts

```diff
@@ -363,19 +364,31 @@ export default class Attribute extends DataColumn<AttributeOptions, AttributeInt
     return result;
   }
 
-  getBufferLayout(): BufferLayout {
+  /** Generate WebGPU-style buffer layout descriptor from this attribute */
+  getBufferLayout(
+    /** A luma.gl Model-shaped object that supplies additional hint to attribute resolution */
+    modelInfo?: {isInstanced?: boolean}
+  ): BufferLayout {
     // Clear change flag
     this.state.layoutChanged = false;
 
     const shaderAttributeDefs = this.settings.shaderAttributes;
-    const result: BufferLayout = super.getBufferLayout();
+    const result: BufferLayout = super._getBufferLayout();
+    const {stepMode} = this.settings;
+    if (stepMode === 'dynamic') {
+      // If model info is provided, use isInstanced flag to determine step mode
+      // If no model info is provided, assume it's an instanced model (most common use case)
+      result.stepMode = modelInfo ? (modelInfo.isInstanced ? 'instance' : 'vertex') : 'instance';
```

Comment:
Is it worth logging a warning here if falling back to assumptions? We generally want to reduce the amount of magic we rely on...

Or is it too common / convenient to let layers omit?

Reply:
Every layer calls `attributeManager.getBufferLayouts()` when constructing their models. Even if the layer does not contain any attribute with `stepMode: 'dynamic'`, it may be used with an extension that does. Unless you think `getBufferLayouts` should always be called with a `isInstanced` argument  (which is quite verbose), I'd prefer a quiet fallback.

I am personally not a fan of the noisy luma warnings in v9. Users should not be shown warnings that they cannot do anything about. Log levels should be used for development purpose.

Reply:
> Unless you think getBufferLayouts should always be called with a isInstanced argument (which is quite verbose), I'd prefer a quiet fallback.

Understood. For the longer term. I am suspecting that AttributeManager is perhaps doing too much. One the one had it is a binary table generator and on the other hand it is used to map that binary table to different attribute sets. I suspect that further separating these two steps at some point (e.g. remove `addInstanced()`) might make it easier to generate precise layouts for each model.That is, whether the binary columns in the table is instanced seems like a technicality that the binary table management shouldn't need to worry about, but should be applied when a layout is requested. So in this sense, all attributes should be dynamic. 

I haven't touched this code in a long time so it would probably take some discussion to work it through.

> I am personally not a fan of the noisy luma warnings in v9. Users should not be shown warnings that they cannot do anything about. Log levels should be used for development purpose.

Sure, whether those logs are shown at debug level 0 or level 1 is a reasonable separate discussion. 

Though in my view that doesn't solve anything. Relying on warnings being suppressed doesn't seem right, and showing them in debug mode even though they are "intentional" isn't helpful either, so I still I feel that until we find a way for deck to generate more precise layouts, we still need options for deck to turn them off.